src/Filesystem.cc

```diff
@@ -195,11 +199,12 @@ bool ignition::common::chdir(const std::string &_dir)
 /////////////////////////////////////////////////
 std::string ignition::common::basename(const std::string &_path)
 {
-  fs::path p(_path);
-  // Maintain compatibility with ign-common
-  if (*_path.rbegin() == fs::path::preferred_separator)
-    p = fs::path(_path.substr(0, _path.size()-1));
-  return p.filename().string();
+  fs::path p = fs::path(_path);
+  p /= "FOO.TXT";
+  p = p.lexically_normal();
+  p = p.parent_path();
+  return p.filename().string().empty() ? 
+    std::string{fs::path::preferred_separator} : p.filename().string();
```

Comment:
I see that the `parentPath` function below is using the same logic that's being replaced here. Maybe that needs to be updated too?

Reply:
Well, as it turns out, I wasn't matching behavior correctly.  I opened a PR against `ign-common4` with some more tests to make sure that basename/parentPath were matching between the two versions: #309  and then cherry-picked it here.